src/tests/src/solver/expressions/test_expressions.cpp

```diff
@@ -428,14 +428,68 @@ BOOST_FIXTURE_TEST_CASE(comparison_nodes_are_not_linear, Registry<Node>)
     // x==y
     Node* eq = create<EqualNode>(&var1, &var2);
     BOOST_CHECK_EQUAL(printVisitor.dispatch(*eq), "x==y");
-    BOOST_CHECK_EQUAL(linearVisitor.dispatch(*eq), LinearStatus::NON_LINEAR);
+    BOOST_CHECK_EQUAL(linearVisitor.dispatch(*eq), LinearStatus::LINEAR);
     // x<=y
     Node* lt = create<LessThanOrEqualNode>(&var1, &var2);
     BOOST_CHECK_EQUAL(printVisitor.dispatch(*lt), "x<=y");
-    BOOST_CHECK_EQUAL(linearVisitor.dispatch(*lt), LinearStatus::NON_LINEAR);
+    BOOST_CHECK_EQUAL(linearVisitor.dispatch(*lt), LinearStatus::LINEAR);
     // x>=y
     Node* gt = create<GreaterThanOrEqualNode>(&var1, &var2);
     BOOST_CHECK_EQUAL(printVisitor.dispatch(*gt), "x>=y");
+    BOOST_CHECK_EQUAL(linearVisitor.dispatch(*gt), LinearStatus::LINEAR);
+}
+
+BOOST_FIXTURE_TEST_CASE(comparison_nodes_variable_constant_is_linear, Registry<Node>)
+{
+    PrintVisitor printVisitor;
+    LinearityVisitor linearVisitor;
+
+    VariableNode var1("x");
+    // variable
+    LiteralNode literal(21.);
+    // x==y
```

Comment:
```C++
// x==y
```
?
Do we mean : 
```C++
// x=21
```
instead ?
(I didn't know we could meet a '=='  in an expression, could we ?)

Reply:
yes